### PR TITLE
[All Hosts] (coding) include file about how to support IE with post-E…

### PR DESCRIPTION
**Pull request by Rick-Kirkham**
…S5 JavaScript

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[d1f90bc](https://github.com/OfficeDev/office-js-docs-pr/commits/d1f90bc582bf4eb37ac3d167a9fe0d0dcdfbf308)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/concepts/add-in-development-best-practices.md](https://github.com/OfficeDev/office-js-docs-pr/blob/es5-support/docs/concepts/add-in-development-best-practices.md) | :white_check_mark:Succeeded | |
[docs/develop/common-coding-issues.md](https://github.com/OfficeDev/office-js-docs-pr/blob/es5-support/docs/develop/common-coding-issues.md) | :white_check_mark:Succeeded | |
[docs/includes/es5-support.md](https://github.com/OfficeDev/office-js-docs-pr/blob/es5-support/docs/includes/es5-support.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C4%5C23%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202004230019352462-1782%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=XM1fCxPEyD%2Fuvn5SUCOU0oR9JqlFpp%2BgQ05HlP57DYM%3D&st=2020-04-23T00%3A15%3A48Z&se=2020-05-24T00%3A20%3A48Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/includes/es5-support.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+For some versions of Office and Windows, the JavaScript engine in which add-ins run is provided by Internet Explorer. (For details of which versions, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).) The Internet Explorer engine does not support versions of JavaScript later than ES5. This means that without special handling the JavaScript files that your add-in serves cannot use syntax, types, or methods that were added to the language after ES5. This does not mean that you must *write* in ES5 syntax. You have two other options:
+
+- Write your code in ECMAScript 2015 (also called ES6) or later JavaScript, or in TypeScript, and then compile your code to ES5 JavaScript using a compiler such as [babel](https://babeljs.io/) or [tsc](https://www.typescriptlang.org/index.html).
+- Write in ECMAScript 2015 or later JavaScript, but also load a [polyfill](https://en.wikipedia.org/wiki/Polyfill_(programming)) library such as [core-js](https://github.com/zloirock/core-js) that enables IE to run your code.
```
Comment (AlexJerabek) [inline docs/includes/es5-support.md:4]: Link to the non-localized wiki page:
```suggestion
- Write in ECMAScript 2015 or later JavaScript, but also load a [polyfill](https://wikipedia.org/wiki/Polyfill_(programming)) library such as [core-js](https://github.com/zloirock/core-js) that enables IE to run your code.
```

File: docs/includes/es5-support.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+For some versions of Office and Windows, the JavaScript engine in which add-ins run is provided by Internet Explorer. (For details of which versions, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).) The Internet Explorer engine does not support versions of JavaScript later than ES5. This means that without special handling the JavaScript files that your add-in serves cannot use syntax, types, or methods that were added to the language after ES5. This does not mean that you must *write* in ES5 syntax. You have two other options:
```
Comment (AlexJerabek) [inline docs/includes/es5-support.md:1]: Consider adding a comma after "special handling" to help readers parse out the conditional. 
```suggestion
For some versions of Office and Windows, the JavaScript engine in which add-ins run is provided by Internet Explorer. (For details of which versions, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).) The Internet Explorer engine does not support versions of JavaScript later than ES5. This means that without special handling, the JavaScript files that your add-in serves cannot use syntax, types, or methods that were added to the language after ES5. This does not mean that you must *write* in ES5 syntax. You have two other options:
```

Comment (AlexJerabek) [inline docs/includes/es5-support.md:1]: I'm not sure this follows MSFT style guidelines, but I'd find this to be more readable if the link to the "Browsers used" article was in the first sentence:
"For [some versions of Office and Windows](../concepts/browsers-used-by-office-web-add-ins.md), the JavaScript engine in which add-ins run is provided by Internet Explorer."

Comment (AlexJerabek) [inline docs/includes/es5-support.md:1]: Is there an easy external resource we can link people to about what's in ES6+?

**Comment by o365devx** [comment]
Docs Build status updates of commit _[14456d3](https://github.com/OfficeDev/office-js-docs-pr/commits/14456d3cf15c954e8f53e91cf0b55141228044a9)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/concepts/add-in-development-best-practices.md](https://github.com/OfficeDev/office-js-docs-pr/blob/es5-support/docs/concepts/add-in-development-best-practices.md) | :white_check_mark:Succeeded | |
[docs/develop/common-coding-issues.md](https://github.com/OfficeDev/office-js-docs-pr/blob/es5-support/docs/develop/common-coding-issues.md) | :white_check_mark:Succeeded | |
[docs/includes/es5-support.md](https://github.com/OfficeDev/office-js-docs-pr/blob/es5-support/docs/includes/es5-support.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C4%5C23%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202004231750408310-1782%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=z01vhgN%2BVTvgoP0t7%2BX2UM8fn47IM0HTsTbvnkODR5g%3D&st=2020-04-23T17%3A47%3A09Z&se=2020-05-24T17%3A52%3A09Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[21e317c](https://github.com/OfficeDev/office-js-docs-pr/commits/21e317c52d8900f8713be2ed72344b28f55af26b)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/concepts/add-in-development-best-practices.md](https://github.com/OfficeDev/office-js-docs-pr/blob/es5-support/docs/concepts/add-in-development-best-practices.md) | :white_check_mark:Succeeded | |
[docs/develop/common-coding-issues.md](https://github.com/OfficeDev/office-js-docs-pr/blob/es5-support/docs/develop/common-coding-issues.md) | :white_check_mark:Succeeded | |
[docs/includes/es5-support.md](https://github.com/OfficeDev/office-js-docs-pr/blob/es5-support/docs/includes/es5-support.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C4%5C23%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C202004231758071814-1782%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=s3CCy1fcWoqnD7ro9CBq6nB8GHw3TJvACEJSU2jW5D8%3D&st=2020-04-23T17%3A54%3A20Z&se=2020-05-24T17%3A59%3A20Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**